### PR TITLE
Add support for compilation on MacOS

### PR DESCRIPTION
**Pull request by eloparco**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -51,4 +51,4 @@ benchmark.wamr: random.bin $(PIGZ_WASM) $(WAMR)
 random.bin: random.original.bin
 	@cp $< $@
 random.original.bin:
-	head -c 100M </dev/urandom >$@
+	head -c 100000000 </dev/urandom >$@
```
Comment (eloparco) [inline Makefile]: `100M` is not supported on MacOS

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -25,3 +25,18 @@ jobs:
       run: make benchmark
     - name: Run benchmark - WAMR
       run: make benchmark.wamr
+  check-macos:
```
Comment (loganek) [inline .github/workflows/main.yml]: you could probably avoid duplicating the code here and use a [matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) instead:
```yml
 strategy:
      matrix:
        platform: [linux, macos]
```
and then use it in the job itself:
```yml
    - name: Build components
      run: make PLATFORM=${{ matrix.platform }}
```

File: Makefile (hunk 1 above)
Comment (loganek) [reply Makefile]: Technically 100M = 104857600 bytes, but I guess it doesn't really matter here.

File: README.md [hunk 3]
```diff
@@ -2,7 +2,7 @@ wasm-parallel-gzip
 ===================
 
 This project demonstrates how to build and run parallel compression using WebAssembly and
-[wasi-threads]. It compiles the [pigz] CLI tool to WebAssembly using a [wasi-sdk] pre-release and
+[wasi-threads]. It compiles the [pigz] CLI tool to WebAssembly using a [wasi-sdk] release and
 runs it using a recent version of [wasmtime] (and optionally, [wamr]). At the moment, wasi-threads
```
Comment (loganek) [inline README.md:6]: looks like this is no longer optional based on your changes in `Makefile`?

Comment (eloparco) [reply README.md:6]: Should I just update the readme or revert it to optional (by removing the WAMR dependency from the `build` target, as it was before)?

Comment (loganek) [reply README.md:6]: I don't have a strong opinion on that, I think either way is fine. We can probably make the WAMR optional given that compiling WAMR requires third party dependencies that are not installed automatically (e.g. CMake)

**Review by loganek — APPROVED** [review]
LGTM, thanks

**Comment by abrown** [comment]
@eloparco, thanks for the PR. I'll take a closer look later today or tomorrow.

**Review by abrown — APPROVED** [review]
LGTM! Thanks for de-duplicating the CI configuration in `main.yml`.

**Pull request merged**